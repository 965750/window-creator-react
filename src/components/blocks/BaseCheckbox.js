import React from 'react'
import styles from './BaseCheckbox.module.scss'

const BaseCheckbox = ({ isChecked, classes }) => {
    return (
        <div
            className={`${styles.checkbox__wrapper} border-botticelli border-4 rounded-full relative ${classes}`}
        >
            {isChecked ? (
                <div
                    className={`${
                        styles['checkbox--checked']
                    } top-half left-half rounded-full bg-hippieBlue absolute`}
                />
            ) : null}
        </div>
    )
}

export default BaseCheckbox
