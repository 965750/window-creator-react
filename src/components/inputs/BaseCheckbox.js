import PropTypes from 'prop-types'
import React from 'react'
import styles from './BaseCheckbox.module.scss'

const BaseCheckbox = ({ isChecked, classes }) => (
  <div
    className={`${styles.checkbox__wrapper} border-gullGray border relative ${classes}`}
  >
    {isChecked ? (
      <div
        className={`${
          styles['checkbox--checked']
        } border-b-4 border-r-4 border-hippieBlue top-half left-half absolute`}
      />
    ) : null}
  </div>
)

BaseCheckbox.propTypes = {
  isChecked: PropTypes.bool.isRequired,
  classes: PropTypes.string,
}

export default BaseCheckbox
