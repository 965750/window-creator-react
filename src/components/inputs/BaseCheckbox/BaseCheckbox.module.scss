.checkbox {

  &__wrapper {
    width: 23px;
    height: 23px;
  }

  &--checked {
    width: 9px;
    height: 15px;
    transform: translate(-50%, -60%) rotate(30deg);
  }
}