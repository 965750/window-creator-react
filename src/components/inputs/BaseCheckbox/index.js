import PropTypes from 'prop-types'
import React from 'react'
import styles from './BaseCheckbox.module.scss'

const BaseCheckbox = ({ isChecked, classes }) => (
  <div
    data-testid="checkboxWrapper"
    className={`${styles.checkbox__wrapper} border-gullGray border relative ${classes}`}
  >
    {isChecked ? (
      <div
        data-testid="checkboxIcon"
        className={`${
          styles['checkbox--checked']
        } border-b-4 border-r-4 border-hippieBlue top-half left-half absolute`}
      />
    ) : null}
  </div>
)

BaseCheckbox.propTypes = {
  isChecked: PropTypes.bool.isRequired,
  classes: PropTypes.string,
}

export default BaseCheckbox
