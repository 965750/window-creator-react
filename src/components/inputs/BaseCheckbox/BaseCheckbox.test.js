import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import BaseCheckbox from './index'
import messages from '../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('BaseCheckbox', () => {
  let wrapper
  const initialProps = {
    isChecked: false,
    classes: ''
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <BaseCheckbox {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot with unchecked checkbox', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should have icon inside div when checkbox is checked', () => {
    setUp({
      isChecked: true,
      classes: ''
    })

    expect(wrapper.queryByTestId('checkboxIcon')).not.toBeNull()
  })

  it('should pass classes from props to wrapper', () => {
    const mockClasses = 'mockClasses'

    setUp({
      isChecked: true,
      classes: mockClasses
    })

    expect(wrapper.queryByTestId('checkboxWrapper')).toHaveClass(mockClasses)
  })
})
