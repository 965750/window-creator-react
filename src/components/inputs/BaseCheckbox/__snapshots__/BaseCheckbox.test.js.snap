// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCheckbox should match snapshot with unchecked checkbox 1`] = `
<DocumentFragment>
  <div
    class="checkbox__wrapper border-gullGray border relative "
    data-testid="checkboxWrapper"
  />
</DocumentFragment>
`;
